src/menu/Menu.ts

```diff
@@ -0,0 +1,299 @@
+
+		// Assuming the menu has an `auto` height, manually apply the scroll height (or max-height if specified),
+		// and animate to and from that.
+		requestAnimationFrame(() => {
+			const height = getMenuHeight(element);
+			this.wasOpen = !hidden;
+
+			if (hidden) {
+				// Temporarily remove any transition to prevent the element from animating to `height`.
+				const transition = element.style.transition;
+				element.style.transition = null;
+
+				element.style.height = height + 'px';
+				element.style.transition = transition;
+				requestAnimationFrame(() => {
+					element.style.height = '0';
+				});
+			}
+			else {
+				element.scrollTop = 0;
+				element.style.height = `${height}px`;
+			}
+		});
+	}
+
+	protected getDefaultHidden() {
+		const { disabled, label } = this.properties;
+
+		if (label && disabled) {
+			return true;
+		}
+
+		return label ? true : false;
+	}
+
+	protected getMenuClasses() {
+		const { animate = true, hidden = this.getDefaultHidden(), label, nested } = this.properties;
+		const isSubMenu = Boolean(label);
+		const classes = [ css.menu ];
+
+		if (this._initialRender || !animate || !isSubMenu) {
+			classes.push(hidden ? css.hidden : css.visible);
+		}
+
+		if (nested) {
+			classes.push(css.nestedMenu);
+		}
+
+		if (isSubMenu) {
+			classes.push(css.subMenu);
+		}
+
+		return classes;
+	}
+
+	protected onElementCreated(element: HTMLElement, key: string) {
+		if (key === 'menu') {
+			const { animate = true, hidden = this.getDefaultHidden() } = this.properties;
+			this._initialRender = false;
+
+			if (animate) {
+				this.wasOpen = !hidden;
+
+				if (hidden) {
+					element.style.height = '0';
+				}
+			}
+		}
+	}
+
+	protected onElementUpdated(element: HTMLElement, key: string) {
+		if (key === 'menu') {
+			const { animate = true, label } = this.properties;
+
+			if (!label) {
+				return;
+			}
+
+			if (!animate) {
+				// In case `animate` was previously `true`, remove any `height` property set on the node.
+				element.style.height = null;
+				return;
+			}
+
+			this.animate(element);
+		}
+	}
+
+	protected onLabelClick() {
+		const {
+			disabled,
+			expandOnClick = true
+		} = this.properties;
+
+		if (!disabled && expandOnClick) {
+			this.toggleDisplay();
+		}
+	}
+
+	protected onLabelKeypress(event: KeyboardEvent) {
+		const { disabled } = this.properties;
+		const key: string | number = 'key' in event ? event.key : event.keyCode;
+
+		if (!disabled && (key === 'Enter' || key === 13)) {
+			this.toggleDisplay();
+		}
+	}
+
+	protected onMenuFocus() {
+		const { disabled, hidden = this.getDefaultHidden() } = this.properties;
+		if (!disabled && hidden) {
+			this.toggleDisplay(true);
+		}
+	}
+
+	protected onMenuMouseEnter() {
+		const {
+			disabled,
+			expandOnClick = true
+		} = this.properties;
+
+		if (!disabled && !expandOnClick) {
+			this._hideTimer && this._hideTimer.destroy();
+			this.toggleDisplay(true);
+		}
+	}
```

Comment:
When the dropdown can be opened on hover, we should probably consider having it open on focus as well